### PR TITLE
NEWRELIC-5565 add versioned tests for Fastify CLM

### PR DESCRIPTION
**Pull request by jmartin4563**
## Proposed Release Notes
* Added CLM versioned tests for Fastify

## Links
* Closes NEWRELIC-5565

## Details
* Fastify@3 requires the use of `middie` for middleware registration with `use`, so there are two test files: one that uses `middie` for >3, and a 2.x version that does not use `middie`
* Added hook based tests to ensure CLM coverage in both styles of Fastify route handling

### CODE REVIEW

**Comment by kibertoad** [comment]
Why not include tests that are using modern fastify lifecycle hooks via plugins, without middleware, too?

**Comment by bizob2828** [comment]
> Why not include tests that are using modern fastify lifecycle hooks via plugins, without middleware, too?

We talked about this offline. It'll be happening, are you interested in eventually using this feature @kibertoad? There's work to be done on a different team to actually get the integration with CodeStream but always looking for real users to give feedback

**Comment by kibertoad** [comment]
@bizob2828 We are using both NewRelic and fastify in Lokalise, but I don't think we use CLM yet. It sounds interesting, though, so I'm definitely up for trying it.

**Review by bizob2828 — DISMISSED** [review]
overall looks great, one food for thought

File: test/versioned/fastify/code-level-metrics-middleware.tap.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+const common = require('./common')
+
+tap.Test.prototype.addAssert('clmAttrs', 1, helper.assertCLMAttrs)
+
+async function setupFastifyServer(fastify, calls) {
+  common.setupRoutes(fastify)
+  await fastify.register(require('middie'))
```
[the remark this answers is not in the document]
Comment (jmartin4563) [reply test/versioned/fastify/code-level-metrics-middleware.tap.js:16]: The `clmAttrs` segment assertions are also different between the v2 and v3 suite for middleware based routes, but I can still combine the two and just have v2 and v3 assertion vars, and v2 and v3 setup functions, no biggie!

[the remark this answers is not in the document]
Comment (bizob2828) [reply test/versioned/fastify/code-level-metrics-middleware.tap.js:16]: ok, I'm just thinking of any ways to make this easier to maintain

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**